File: examples/routines/procedure.f90
subroutine f(x, xsquared)
    implicit none

    real, intent(in) :: x
    real, intent(out) :: xsquared
    
    xsquared = x*x

end subroutine f

program function_example
    implicit none
    
    real :: x = 1.5
    real :: xsquared = 0.0

    write(*,*) "Fortran procedure example"
    call f(x, xsquared)
    write(*,*) x, " squared is ", xsquared
    
end program function_example

File: examples/routines/procedure.cpp
#include<iostream>


void f(double x, double& xsquared) {
    xsquared = x*x;
}

int main() {
    
    std::cout << "C++ Example of procedure" << std::endl;
    double x = 1.5;
    double xsquared = 0;
    f(x, xsquared);
    std::cout << x << " squared is " << xsquared << std::endl;
}
